c7n/resources/ecr.py

```diff
@@ -63,6 +63,49 @@ def process_resource_set(self, client, resources, tags):
                 pass
 
 
+@ECR.action_registry.register('set-scanning')
+class ECRSetScanning(Action):
+
+    permissions = ('ecr:PutImageScanningConfiguration',)
+    schema = type_schema(
+        'set-scanning',
+        state={'type': 'boolean', 'default': True})
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('ecr')
+        s = self.data.get('state', True)
+        for r in resources:
+            try:
+                client.put_image_scanning_configuration(
+                    registryId=r['registryId'],
+                    repositoryName=r['repositoryName'],
+                    imageScanningConfiguration={
+                        'scanOnPush': s})
+            except client.exceptions.RepositoryNotFoundException:
+                continue
+
+
+@ECR.action_registry.register('set-immutability')
+class ECRSetImmutability(Action):
+
+    permissions = ('ecr:PutImageTagMutability',)
+    schema = type_schema(
+        'set-immutability',
+        state={'type': 'boolean', 'default': True})
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('ecr')
+        s = self.data.get('state', True) and 'IMMUTABLE' or 'MUTABLE'
```

Comment:
I know it works, but it looks really weird.. Should it be ternary operator instead?

Reply:
lol, I guess I'm old.. to me this is the classic syntax for ternary style operators, but its pretty dated (boolean expression syntax), I can update to the newer syntax (py2.4+) for conditional expressions. thanks for looking.